src/Disks/IO/CachedOnDiskReadBufferFromFile.cpp

```diff
@@ -857,7 +854,7 @@ bool CachedOnDiskReadBufferFromFile::nextImplStep()
         implementation_buffer = getImplementationBuffer(file_segments->front());
 
         if (read_type == ReadType::CACHED)
-            file_segments->front().incrementHitsCount();
+            file_segments->front().use();
```

Comment:
there are to places where use_count is incremented 
1. here
2. in completeFileSegmentAndGetNext

in (2) it was ok, in (1) it was not. (only method `use()` changes actual use count).

TODO: will make a unit test. (currently use_count is tested with unit tests, but without CachedOnDiskReadBufferFromFile, e.g. tested on FileCache api level)